### PR TITLE
Xapid-717

### PR DESCRIPTION
**Pull request by theganyo**
status code checks, logging of errors, client back off & timeouts

### CODE REVIEW

File: apigee_sync.go [hunk 1]
```diff
@@ -52,41 +52,33 @@ func postPluginDataDelivery(e apid.Event) {
...
  */
 func updatePeriodicChanges() {
 
-	times := 1
-	pollInterval := config.GetInt(configPollInterval)
+	var backOffFunc func()
+	pollInterval := config.GetDuration(configPollInterval)
 	for {
-		startTime := time.Second
+		start := time.Now().Second()
```
Comment (haomingzhang) [inline apigee_sync.go]: In time.go:
"Second returns the second offset within the minute specified by t, in the range [0, 59]."
Maybe we should use time.now().Unix(), or just use time.Now() with duration comparison.

File: apigee_sync.go [hunk 2]
```diff
@@ -464,6 +461,11 @@ func downloadSnapshot() {
 			continue
 		}
 
+		if r.StatusCode != 200 {
+			log.Errorf("Snapshot server conn failed. HTTP Resp code %d", r.StatusCode)
+			continue
```
Comment (haomingzhang) [inline apigee_sync.go]: r.Body.Close()

File: apigee_sync.go [hunk 3]
```diff
@@ -464,6 +461,11 @@ func downloadSnapshot() {
 			continue
```
Comment (haomingzhang) [inline apigee_sync.go]: r.Body.Close()

File: apigee_sync.go (hunk 1 above)
Comment (theganyo) [inline apigee_sync.go]: Fixed in PR-23.

File: apigee_sync.go (hunk 2 above)
Comment (theganyo) [inline apigee_sync.go]: Fixed in PR-23

**Pull request merged**